zebra-scan/src/tests.rs

```diff
@@ -550,3 +509,47 @@ fn random_compact_tx(mut rng: impl RngCore) -> CompactTx {
     ctx.outputs.push(cout);
     ctx
 }
+
+/// Returns transactions belonging to any of the given [`ScanningKey`]s.
+///
+/// TODO:
+/// - Remove the `sapling_tree_size` parameter or turn it into an `Option` once we have access to
+/// Zebra's state, and we can retrieve the tree size ourselves.
+/// - Add prior block metadata once we have access to Zebra's state.
+fn scan_block<K: ScanningKey>(
+    network: Network,
+    block: Arc<Block>,
+    sapling_tree_size: u32,
+    scanning_keys: &[&K],
+) -> Result<ScannedBlock<K::Nf>, ScanError> {
+    let network: zcash_primitives::consensus::Network = match network {
+        Network::Mainnet => zcash_primitives::consensus::Network::MainNetwork,
+        Network::Testnet => zcash_primitives::consensus::Network::TestNetwork,
+    };
```

Comment:
Optional:

I was pretty sure we already had code for this but it seems i got confused with something else. We might want to move this code to a separated function in the future.

Reply:
This conversion is universal, so it should go in `zebra-chain`. Otherwise we will end up copying this code around a lot, and it will be harder to modify in tickets like #7968.

A good place would be `zebra-chain/src/primitives/zcash_primitives.rs`.

This matches how we do the conversion for `zcash_address::Network`:
https://github.com/ZcashFoundation/zebra/blob/f08cc2dc50a8620fcbeef48808c53de353e80983/zebra-chain/src/primitives/address.rs#L43-L63

Reply:
Done in [e1735b5](https://github.com/ZcashFoundation/zebra/pull/7994/commits/e1735b5b622155bfe24fb576abff0c21fc718365).